core/server/worker/src/test/java/alluxio/worker/page/PagedBlockWriterTest.java

```diff
@@ -0,0 +1,150 @@
+
+  @Parameterized.Parameter(2)
+  public int mPageSize;
+
+  private LocalCacheManager mCacheManager;
+  private InstancedConfiguration mConf = Configuration.copyGlobal();
+  private PageMetaStore mPageMetaStore;
+  private CacheEvictor mEvictor;
+  private LocalPageStoreOptions mPageStoreOptions;
+  private PageStore mPageStore;
+  private LocalPageStoreDir mPageStoreDir;
+  private PagedBlockWriter mWriter;
+
+  @Rule
+  public TemporaryFolder mFolder = new TemporaryFolder();
+
+  @Rule
+  public ExpectedException mThrown = ExpectedException.none();
+
+  @Before
+  public void before() throws Exception {
+    mConf.set(PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE, mPageSize);
+    mPageMetaStore = new DefaultPageMetaStore();
+    mPageStoreOptions = (LocalPageStoreOptions) PageStoreOptions.create(mConf).get(0);
+    mPageStore = PageStore.create(mPageStoreOptions);
+    mEvictor = new FIFOCacheEvictor(mConf);
+    mPageStoreDir = new LocalPageStoreDir(mPageStoreOptions, mPageStore, mEvictor);
+    mCacheManager =
+        LocalCacheManager.create(mConf, mPageMetaStore, ImmutableList.of(mPageStoreDir));
+    CommonUtils.waitFor("restore completed",
+        () -> mCacheManager.state() == CacheManager.State.READ_WRITE,
+        WaitForOptions.defaults().setTimeoutMs(10000));
+    mWriter = new PagedBlockWriter(mCacheManager, BLOCK_ID, mPageSize);
+  }
+
+  @After
+  public void after() throws Exception {
+    mWriter.close();
+  }
+
+  @Test
+  public void appendByteBuf() throws Exception {
+    for (int offset = 0; offset < mFileLength; offset += mChunkSize) {
+      int bytesToWrite = Math.min(mChunkSize, mFileLength - offset);
+      ByteBuf buffer = Unpooled.wrappedBuffer(
+          BufferUtils.getIncreasingByteBuffer(bytesToWrite));
+      assertEquals(bytesToWrite, mWriter.append(buffer));
+    }
+    mWriter.close();
+    verifyDataInCache();
+  }
+
+  @Test
+  public void append() throws Exception {
+    for (int offset = 0; offset < mFileLength; offset += mChunkSize) {
+      int bytesToWrite = Math.min(mChunkSize, mFileLength - offset);
+      ByteBuffer buffer =
+          BufferUtils.getIncreasingByteBuffer(bytesToWrite);
+      assertEquals(bytesToWrite, mWriter.append(buffer));
+    }
+    mWriter.close();
+    verifyDataInCache();
+  }
+
+  private void verifyDataInCache() {
+    List<PageId> pageIds =
+        mCacheManager.getCachedPageIdsByFileId(String.valueOf(BLOCK_ID), mFileLength);
+    assertEquals((int) Math.ceil((double) mFileLength / mPageSize), pageIds.size());
+    byte[] dataInCache = new byte[mFileLength];
+    for (int i = 0; i < pageIds.size(); i++) {
+      PageId pageId = pageIds.get(i);
+      mCacheManager.get(pageId, 0, Math.min(mPageSize, mFileLength - i * mPageSize), dataInCache,
+          i * mPageSize);
+    }
+    for (int offset = 0; offset < mFileLength; offset += mChunkSize) {
+      BufferUtils.equalIncreasingByteArray(offset, Math.min(mChunkSize, mFileLength - offset),
+          dataInCache);
```

Comment:
`equalIncreasingByteArray` returns a boolean that needs to be checked.

```suggestion
      assertTrue(BufferUtils.equalIncreasingByteArray(offset, Math.min(mChunkSize, mFileLength - offset),
          dataInCache));
```

Reply:
`equalIncreasingByteArray` requires that the array has the same length as the second parameter:

> The array length must be equal to the length checked.

Reply:
done, this one really help me to catch a severe bug. Thanks a lot!